backend/src/main/java/meetup/connect/event/EventCreateDto.java

```diff
@@ -1,15 +1,40 @@
 package meetup.connect.event;
 
+import com.fasterxml.jackson.annotation.JsonFormat;
+import jakarta.validation.constraints.NotBlank;
+import jakarta.validation.constraints.NotNull;
+import jakarta.validation.constraints.Size;
+import meetup.connect.common.exception.MeetUpError;
+import meetup.connect.common.exception.MeetUpException;
+import org.springframework.format.annotation.DateTimeFormat;
+
 import java.io.Serializable;
 import java.time.LocalDateTime;
 
 public record EventCreateDto(
-    String name, LocalDateTime dateFrom, LocalDateTime dateTo, String address)
+    @NotBlank @Size(max = 50, message = "Name is too long") String name,
+    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd HH:mm") @NotNull
+        LocalDateTime dateFrom,
+    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd HH:mm") @NotNull
+        LocalDateTime dateTo,
+    @NotBlank @Size(max = 100, message = "Address is too long") String address,
+    @NotNull EventType type)
     implements Serializable {
 
-  public static EventCreateDto fromEntity(Event event) {
-    return new EventCreateDto(
-        event.getName(), event.getDateFrom(), event.getDateTo(), event.getAddress());
+  public EventCreateDto {
+    if (!dateTo.isAfter(dateFrom)) {
+      throw new MeetUpException(MeetUpError.WRONG_DATES);
+    }
+    if ((type.equals(EventType.CASUAL_GET_TOGETHER) || type.equals(EventType.PARTY))
```

Comment:
issue: a possible `NullPointerException`

Reply:
a good practice: what could happen if the type was null?
`(null).equals(EventType.CASUAL_GET_TOGETHER) -> NullPointerException`
It would be best to use equals on a non-nullable value to write null-safe code `EventType.CASUAL_GET_TOGETHER` can't be null, so it should be like that `EventType.CASUAL_GET_TOGETHER.equals(type)`

Reply:
note: I strongly suggest an additional reading about the enum comparison in Java (it's not obvious) https://stackoverflow.com/questions/1750435/comparing-java-enum-members-or-equals